SIGSEGV while running the program that checks runnability for ARM64_NEON

The code used to check ARM64_NEON at https://github.com/p12tic/libsimdpp/blob/9dac213d6965bf57c7accf9dda6d16c3bed0e3ac/cmake/SimdppMultiarch.cmake#L747-L779 fails with `SIGSEGV (Address boundary error)` when executed on armv8.2 CPU:

```
$ g++ ../arm64.c && ./a.out
../arm64.c: In function ‘int main()’:
../arm64.c:22:36: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   22 |         uint32x4_t one = vld1q_u32((uint32_t*)(*p));
      |                                    ^~~~~~~~~~~~~~~
../arm64.c:26:19: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   26 |         vst1q_u32((uint32_t*)(*p), one);
      |                   ^~~~~~~~~~~~~~~
fish: Job 1, './a.out' terminated by signal SIGSEGV (Address boundary error)
```

This leads to `RUNNABLE_ARCHS` to be just `NONE_NULL`.

The [code](https://github.com/p12tic/libsimdpp/blob/9dac213d6965bf57c7accf9dda6d16c3bed0e3ac/cmake/SimdppMultiarch.cmake#L693-L728) used to check `ARM_NEON` works fine.


CPU details:
```
$ lscpu
Architecture:           aarch64
  CPU op-mode(s):       64-bit
  Byte Order:           Little Endian
CPU(s):                 8
  On-line CPU(s) list:  0-7
Vendor ID:              HiSilicon
  Model name:           Kunpeng-920
    Model:              0
    Thread(s) per core: 1
    Core(s) per socket: 8
    Socket(s):          1
    Stepping:           0x1
    Frequency boost:    disabled
    CPU max MHz:        2400.0000
    CPU min MHz:        2400.0000
    BogoMIPS:           200.00
    Flags:              fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma dcpop asimddp asimdfhm
...
```

Same error happens on x86_64 with QEMU emulated ARM64.

```
$ g++ --version 
g++ (GCC) 10.3.1
```

--- a/cmake/SimdppMultiarch.cmake
+++ b/cmake/SimdppMultiarch.cmake
@@ -768,13 +768,13 @@ set(SIMDPP_ARM64_NEON_TEST_CODE
         char* p = data;
         p = prevent_optimization(p);
 
-        uint32x4_t one = vld1q_u32((uint32_t*)(*p));
+        uint32x4_t one = vld1q_u32((uint32_t*)p);
         one = vaddq_u32(one, one);
 
         // GCC 4.8 misses a subset of functions
         one = vdupq_laneq_u32(one, 1);
 
-        vst1q_u32((uint32_t*)(*p), one);
+        vst1q_u32((uint32_t*)p, one);
         p = prevent_optimization(p);
     }"
 )

